src/programQueueProcessor/unit.test.ts

```diff
@@ -170,13 +213,67 @@ describe("kafka integration", () => {
           resolve();
         }, 30000);
       });
+      const totalEsDocuments_1 = (
+        await esClient.search({
+          index: ALIAS_NAME,
+          track_total_hits: true,
+        })
+      ).body?.hits?.total?.value;
+      expect(totalEsDocuments_1).to.equal(DB_COLLECTION_SIZE);
+
+      // 2. update TEST-CA by publishing a clinical event and a RDPC event:
+      await programQueueProcessor.enqueueEvent({
+        programId: TEST_CA,
+        type: programQueueProcessor.knownEventTypes.CLINICAL,
+      });
+      await programQueueProcessor.enqueueEvent({
+        programId: TEST_CA,
+        type: programQueueProcessor.knownEventTypes.RDPC,
+        rdpcGatewayUrls: [RDPC_URL],
+      });
+
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          resolve();
+        }, 30000);
+      });
+
+      // check if number of TEST-CA documents is expected:
+      const query_test_ca = esb
+        .requestBodySearch()
+        .query(esb.termQuery("programId", TEST_CA));
+
+      const test_ca_documents = (
+        await esClient.search({
+          index: ALIAS_NAME,
+          body: query_test_ca,
+        })
+      ).body?.hits?.total?.value;
+      expect(test_ca_documents).to.equal(testDonorIds.length);
+
+      // check if the number of TEST-US documents is expected:
+      const query_test_us = esb
+        .requestBodySearch()
+        .query(esb.termQuery("programId", TEST_US));
+
+      const test_us_documents = (
+        await esClient.search({
+          index: ALIAS_NAME,
+          body: query_test_us,
+          track_total_hits: true,
+        })
+      ).body?.hits?.total?.value;
+      expect(test_us_documents).to.equal(DB_COLLECTION_SIZE);
+
```

Comment:
do we have some tests to make sure the update to the documents are reflected properly after subsequent updates? both clinical and rdpc I guess, since we had a problem where updates were not being applied eh. Should be pretty simple to add eh